Sources/SwiftDiagnostics/DiagnosticsFormatter.swift

```diff
@@ -103,12 +103,76 @@ public struct DiagnosticsFormatter {
         }
 
         for diag in diags.dropLast(1) {
-          annotatedSource.append("\(preMessage)├─ \(diag.message)\n")
+          if colorize {
+            annotatedSource.append("\(preMessage)├─ \(self.colorize(diag.diagMessage))\n")
+          } else {
+            annotatedSource.append("\(preMessage)├─ \(diag.message)\n")
+          }
```

Comment:
What do you think about making all the functions on `DiagnosticFormatter` non-`static`. That way `colorize: Bool` could be a property on `DiagnsoticsFormatter`, you could have `func colorizeIfWanted(_ message: DiagnosticMessage)` and this would become the following where we don’t need to repeat the `preMessage` and box drawing character.

```suggestion
          annotatedSource.append("\(preMessage)├─ \(self.colorizeIfRequested(diag.diagMessage))\n")
```

As an added bonus, this would allow callers of the `DiagnosticFormatter` to also control the context size.

The entry to the diagnostic formatter could still be a static function that looks something like the following:

```swift
public static func annotatedSource<SyntaxType: SyntaxProtocol>(tree: SyntaxType, diags: [Diagnostic], contextSize: Int = 2, colorize: Bool = false) -> String {
  let formatter = DiagnosticFormatter(contextSize: contextSize, colorize: colorize)
  return formatter.annotatedSource(tree: tree, diags: diags)
}
```

Reply:
Yeah I think this is a good improvement; the static-ness actually caught me off guard initially but I didn't want to make too many changes so left it where it was. I'll refactor the `DiagnosticFormatter` in another commit to this PR 👍🏻

Reply:
Don't be afraid to refactor all the things if it makes the API or implementation simpler. 😄